drivers/timer/rcar_cmt_timer.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020 IoT.bzh <julien.massot@iot.bzh>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/timer/system_timer.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/rcar_clock_control.h>
+
+#define DT_DRV_COMPAT renesas_rcar_cmt
+
+#define TIMER_IRQ              DT_INST_IRQN(0)
+#define TIMER_BASE_ADDR        DT_INST_REG_ADDR(0)
+#define TIMER_CLOCK_FREQUENCY  DT_INST_PROP(0, clock_frequency)
+
+#define CLOCK_SUBSYS           DT_INST_CLOCKS_CELL(0, module)
+
+#define CYCLES_PER_SEC         TIMER_CLOCK_FREQUENCY
+#define CYCLES_PER_TICK        (CYCLES_PER_SEC / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static struct rcar_cpg_clk mod_clk = {
+	.module = DT_INST_CLOCKS_CELL(0, module),
+	.domain = DT_INST_CLOCKS_CELL(0, domain),
+};
+
+BUILD_ASSERT(CYCLES_PER_TICK > 1,
+	     "CYCLES_PER_TICK must be greater than 1");
+
+#define CMCOR0_OFFSET                   0x018   /* constant register 0 */
+#define CMCNT0_OFFSET                   0x014   /* counter 0 */
+#define CMCSR0_OFFSET                   0x010   /* control/status register 0 */
+
+#define CMCOR1_OFFSET                   0x118   /* constant register 1 */
+#define CMCNT1_OFFSET                   0x114   /* counter 1 */
+#define CMCSR1_OFFSET                   0x110   /* control/status register 1 */
+
+#define CMCLKE                          0xB00   /* CLK enable register */
+#define CLKEN0                          BIT(5)  /* Enable Clock for channel 0 */
+#define CLKEN1                          BIT(6)  /* Enable Clock for channel 1 */
+
+#define CMSTR0_OFFSET                   0x000   /* Timer start register 0 */
+#define CMSTR1_OFFSET                   0x100   /* Timer start register 1 */
+#define START_BIT                       BIT(0)
+
+#define CSR_CLK_DIV_1                   0x00000007
+#define CSR_ENABLE_COUNTER_IN_DEBUG     BIT(3)
+#define CSR_ENABLE_INTERRUPT            BIT(5)
+#define CSR_FREE_RUN                    BIT(8)
+#define CSR_WRITE_FLAG                  BIT(13)
+#define CSR_OVERFLOW_FLAG               BIT(14)
+#define CSR_MATCH_FLAG                  BIT(15)
+
+static void cmt_isr(void *arg)
+{
+	ARG_UNUSED(arg);
+	uint32_t reg_val;
+
+	/* clear the interrupt */
+	reg_val = sys_read32(TIMER_BASE_ADDR + CMCSR0_OFFSET);
+	reg_val &= ~CSR_MATCH_FLAG;
+	sys_write32(reg_val, TIMER_BASE_ADDR + CMCSR0_OFFSET);
+
+	/* Announce to the kernel */
+	sys_clock_announce(1);
+}
+
+/*
+ * Initialize both channels at same frequency,
+ * Set the first one to generates interrupt at CYCLES_PER_TICK.
+ * The second one is used for cycles count, the match value is set
+ * at max uint32_t.
+ */
+int sys_clock_driver_init(const struct device *device)
+{
+	const struct device *clk;
+	uint32_t reg_val;
+	int i, ret;
+
+	ARG_UNUSED(device);
+	clk = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(0));
+	if (!clk) {
```

Comment:
Nit: Implicit type conversion (and MISRA violation)
```suggestion
	if (clk == NULL) {
```

Reply:
fixed